components/ai_chat/core/browser/conversation_driver.cc

```diff
@@ -1165,16 +1207,22 @@ void ConversationDriver::OnPremiumStatusReceived(
     mojom::PremiumInfoPtr premium_info) {
   // Maybe switch to premium model when user is newly premium and on a basic
   // model
-  const bool should_switch_model =
-      features::kFreemiumAvailable.Get() &&
-      // This isn't the first retrieval (that's handled in the constructor)
-      last_premium_status_ != mojom::PremiumStatus::Unknown &&
-      last_premium_status_ != premium_status &&
-      premium_status == mojom::PremiumStatus::Active &&
-      GetCurrentModel().access == mojom::ModelAccess::BASIC;
-  if (should_switch_model) {
-    ChangeModel(features::kAIModelsPremiumDefaultKey.Get());
+  const auto& model = GetCurrentModel();
+
+  if (model.option->is_leo_model()) {
```

Comment:
i was worried about the order of evaluation of each operand here and wanted to deterministically read `...get_leo_model()->` separately. i guess i don't need to worry about it because it's a defined behavior in c++ spec as oppose to function arg evaluation order.

Reply:
changed